src/OrchardCore/OrchardCore.Users.Core/CommonPermissions.cs

```diff
@@ -11,13 +11,23 @@ public class CommonPermissions
         /// </summary>
         public static readonly Permission ManageUsers = new Permission("ManageUsers", "Manage Users", true);
 
+        /// <summary>
+        /// Allow the user to Edit other users.
+        /// </summary>
+        public static readonly Permission EditUsers = new Permission("EditUsers", "Edit Other Users", new[] { ManageUsers }, true);
+
+        /// <summary>
+        /// Allow the user to Delete other users.
+        /// </summary>
+        public static readonly Permission DeleteUsers = new Permission("DeleteUsers", "Delete Other Users", new[] { ManageUsers }, true);
+
         /// <summary>
         /// View users only allows listing or viewing a users profile.
         /// </summary>
-        public static readonly Permission ViewUsers = new Permission("View Users", "View Users", new[] { ManageUsers });
+        public static readonly Permission ViewUsers = new Permission("View Users", "View All Users", new[] { ManageUsers, EditUsers, DeleteUsers });
 
         // Dynamic permission template.
-        private static readonly Permission ManageUsersInRole = new Permission("ManageUsersInRole_{0}", "Manage Users in Role - {0}", new[] { ManageUsers });
+        private static readonly Permission ManageUsersInRole = new Permission("ManageUsersInRole_{0}", "Manage Users in Role - {0}", new[] { EditUsers });
```

Comment:
I am concerned it would allow a less restricted user to change people in specific roles. I'd keep it like before for safety.

Reply:
@sebastienros Thanks for looking at this PR.

`ManageUsers` is a higher permission than `EditUsers` so we should be fine to use `EditUsers`. If we keep it to `ManageUsers`, then it we'll eventually override `EditUsers` and `DeleteUsers`. Perhaps, we should rename "Manage Users in Role - {0}" to "Edit Users in Role - {0}"?

However, currently I think we list all users "weather the user can manage them or not". This may be a problem even before this PR. Also before this PR, the `ManageUsersInRole_{0}` without ViewUser is useless since you can't see the users to even manage them. Also, even when a listed in X role and the logged user does not have ManageUsersInRole_X, he is still able to edit that user. I think this is an Existing security issue with that `ManageUsersInRole_` permission.

**A potential fix:** Add a `UserRoleIndex` that would give store the user to role relation "so we can query it" during listing the users. Now, we could add the following to fix this issues.

 1. Add a new service `IUsersAdminListFilter` that would allow us to query the new `UserRoleIndex` to see if a user can manage or edit 
 2. On [this line](https://github.com/OrchardCMS/OrchardCore/blob/c8b2954e038aaa54963b60bc6cfe193cfea54799/src/OrchardCore.Modules/OrchardCore.Users/Services/DefaultUsersAdminListQueryService.cs#L33) add a call to a new service like this `await _usersAdminListFilters.InvokeAsync((filter, model, query, updater) => filter.FilterAsync(model, query, updater), model, query, updater, _logger);`.

Reply:
The ImpliedBy in line 20 might be removable completely.

Reply:
I removed it. same effect. So we list all users for a user with ViewUsers permission. if a user is able to manage users with in X role, the user will see the edit button.